WIP/calculator.rb

```diff
@@ -0,0 +1,95 @@
+class Calculate
+
+  def initialize(**args)
+    @accumulate = args[:accumulate] || args[:accumulate] == nil
+    @discounts = {}
+  end
+
+  def add_discount(**args)
+
+    name = args[:name]
+    amount = args[:amount]
+    percentage = args[:percentage]
+
+
+    if amount.is_a?(String) || percentage.is_a?(String) || amount.to_i < 0  
```

Comment:
amount and percentage is a string? Should they not be integers?

Reply:
This is a condition to check if a string has been given instead of an integer a standard error will be raised.